pkg/types.go

```diff
@@ -35,13 +35,21 @@ func (scnf ErrSecretsNotFound) Error() string {
 	return fmt.Sprintf("No value for the keys: [%s] exists", strings.Join(scnf.keys, ","))
 }
 
+// NewErrSecretsNotFound creates a new ErrSecretsNotFound error.
 func NewErrSecretsNotFound(keys []string) ErrSecretsNotFound {
 	return ErrSecretsNotFound{keys: keys}
 }
 
-// ErrUnsupportedValue error for unsupported data such as invalid characters or the length of key/values
-type ErrUnsupportedValue struct{}
+// ErrUnsupportedProvider error when a secret client provider identifier is either unknown or unsupported.
```

Comment:
Not used outside this file.  Are it and it's supporting method and factory function still required?

Reply:
No, good catch. I will remove this.